plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractTrinoCatalog.java

```diff
@@ -217,9 +229,46 @@ protected SchemaTableName createMaterializedViewStorageTable(ConnectorSession se
 
         String storageSchema = getStorageSchema(definition.getProperties()).orElse(viewName.getSchemaName());
         SchemaTableName storageTable = new SchemaTableName(storageSchema, storageTableName);
-        List<ColumnMetadata> columns = definition.getColumns().stream()
-                .map(column -> new ColumnMetadata(column.getName(), typeForMaterializedViewStorageTable(typeManager.getType(column.getType()))))
-                .collect(toImmutableList());
+
+        Schema schemaWithTimestampTzPreserved = schemaFromMetadata(mappedCopy(
+                definition.getColumns(),
+                column -> {
+                    Type type = typeManager.getType(column.getType());
+                    if (type instanceof TimestampWithTimeZoneType timestampTzType && timestampTzType.getPrecision() <= 6) {
```

Comment:
It looks like we're going to use TIMESTAMP_TZ_MICROS for primitive type but if the same type is in a nested column then we're going to use VARCHAR.
Could you please add an explanation about why we need to do that ?

Reply:
we use varchar for all timestamp with time zone values, unless they are used for partitioning

Reply:
Okay, so is this assuming that only non-nested types are allowed for partitioning ? Will this change when the other PR to allow partitioning on struct types lands ?